src/utils/file/file_utils.cpp

```diff
@@ -0,0 +1,247 @@
+            flags &= ~O_DIRECT;
+            fd = open(path.c_str(), flags, permissions);
+        }
+
+        if (fd == -1) {
+            NIXL_ERROR << absl::StrFormat("Failed to open file: %s - %s", path, strerror(errno));
+            switch (errno) {
+                case EACCES:
+                    return NIXL_ERR_NOT_ALLOWED;
+                case EEXIST:
+                    return NIXL_ERR_INVALID_PARAM;
+                case ENOENT:
+                    return NIXL_ERR_NOT_FOUND;
+                default:
+                    return NIXL_ERR_BACKEND;
+            }
+        }
+    }
+
+    NIXL_DEBUG << absl::StrFormat("Successfully opened file: %s with fd: %d", path, fd);
+    return NIXL_SUCCESS;
+}
+
+nixl_status_t nixlFileUtils::openPrefixedFile(const std::string& prefixed_path, int& fd) {
+    NIXL_DEBUG << absl::StrFormat("Attempting to open prefixed path: %s", prefixed_path);
+
+    auto [success, mode, path] = parsePrefixedPath(prefixed_path);
+    if (!success) {
+        // Error already logged in parsePrefixedPath
+        return NIXL_ERR_INVALID_PARAM;
+    }
+
+    // Check if this is a direct I/O prefix (3-letter prefix ending with 'D')
+    bool use_direct = prefixed_path.length() >= 4 && prefixed_path[2] == 'D' && prefixed_path[3] == ':';
+
+    NIXL_DEBUG << absl::StrFormat("Opening with mode: %d, path: %s, direct I/O: %s",
+                                 static_cast<int>(mode), path, use_direct ? "Yes" : "No");
+
+    return openFile(path, mode, fd, use_direct);
+}
+
+std::tuple<bool, nixlFileMode, std::string> nixlFileUtils::parsePrefixedPath(const std::string& prefixed_path) {
+    // Minimum valid path is "XX:/path" (5 chars)
+    if (prefixed_path.empty()) {
+        NIXL_ERROR << "Invalid path: empty string";
+        return {false, nixlFileMode::READ_ONLY, ""};
+    }
+
+    // Find the colon separator
+    size_t colon_pos = prefixed_path.find(':');
+    if (colon_pos == std::string::npos) {
+        NIXL_ERROR << absl::StrFormat("Invalid path format: missing ':' in '%s'", prefixed_path);
+        return {false, nixlFileMode::READ_ONLY, ""};
+    }
+
+    // Extract and validate prefix
+    std::string prefix = prefixed_path.substr(0, colon_pos);
+    if (prefix.empty() || (prefix.length() != 2 && prefix.length() != 3)) {
+        NIXL_ERROR << absl::StrFormat("Invalid prefix length: '%s' (must be 2 or 3 characters)", prefix);
+        return {false, nixlFileMode::READ_ONLY, ""};
+    }
+
+    // Convert prefix to uppercase for comparison
+    std::transform(prefix.begin(), prefix.end(), prefix.begin(), ::toupper);
+
+    // Extract path (everything after the colon)
+    std::string path = prefixed_path.substr(colon_pos + 1);
+    if (path.empty()) {
+        NIXL_ERROR << absl::StrFormat("Invalid path: no path after prefix '%s:'", prefix);
+        return {false, nixlFileMode::READ_ONLY, ""};
+    }
+
+    // Map prefix to file mode
+    nixlFileMode mode;
+    if (prefix == "RD") {
```

Comment:
where are these prefixes defined and documented? is this part of user-facing API?
can we use unordered_map here?

Reply:
These will be part of optional metadata that user can specify
This was supposed to be part of a separate patch, came in here due to rebase issue. Nevertheless, it hasn't been documented yet, but will be introduced for file operations during register/deregister